.github/copilot-instructions.md

```diff
@@ -107,3 +172,81 @@ try {
     // Code here
 }
 ```
+
+## Error and Warning Patterns
+
+### Error Codes
+- Use `XA####` for MSBuild errors (e.g., `XA1234`)
+- Use `XA####` for MSBuild warnings  
+- Use `APT####` for Android Asset Packaging Tool errors
+- Include clear, actionable error messages
+- Reference documentation when available
+
+### Logging
+- Use MSBuild logging (`Log.LogError`, `Log.LogWarning`, `Log.LogMessage`)
+- Every `LogCodedWarning` and `LogCodedError` needs an error/warning code
+- Include relevant context (file paths, line numbers when available)
+- Make error messages helpful for developers
+
+## Documentation Patterns
+
+### Code Documentation
+- Use XML documentation comments for public APIs
+- Document Android API level requirements where relevant
+- Include `<example>` tags for complex APIs
+- Reference official Android documentation where helpful
+
+### Project Documentation
+- Update relevant `.md` files in `Documentation/` when making significant changes
+- Follow existing documentation structure and formatting
+- Include code examples that actually work
+
+## Commit Message Guidelines
+
+Follow the patterns in `Documentation/workflow/commit-messages.md`:
+
+### Summary Format
+```
+[Component] Summary
+```
+
+Where Component is either:
+- Directory name (e.g., `[Mono.Android]`, `[Build.Tasks]`)
+- Broad category: `build`, `ci`, `docs`, `tests`
+
+### Dependency Bumps
+```
+Bump to org/repo/branch@commit
+Bump to [Dependency Name] [Dependency Version]
+```
+
+### Required Sections
+- **Changes**: What was modified
+- **Fixes**: Issues resolved (include GitHub issue numbers)
+- **Context**: Why the change was needed
+
+## Common Troubleshooting
+
+### Build Issues
+- Clean `bin/` and `obj/` directories
+- Ensure Android SDK/NDK are properly configured
+- Use `make clean` for complete rebuild
+
+### MSBuild Task Development
+- Test tasks in isolation first
+- Handle incremental build scenarios
+- Consider cross-platform compatibility (Windows/macOS/Linux)
+- Validate inputs and provide clear error messages
+
+### Native Development
+- Use appropriate CMake patterns from existing code
+- Handle different Android ABIs (arm64-v8a, x86_64)
```

Comment:
Add back the 32-bit ABIs here: armeabi-v7a and x86. They are still targetted by the Mono native runtime.